contracts/Proofs.sol

```diff
@@ -161,5 +184,19 @@ contract Proofs {
     missed[id] += 1;
   }
 
+  /// @notice Extends the proof end time
+  /// @dev The id must have a mapping to an end id, the end must exist, and the end must not have elapsed yet
+  /// @param id the id of the proofs to extend. Typically a slot id, the id is mapped to an endId.
+  /// @param ending the new end time (in seconds)
+  function _extendProofEndTo(bytes32 id, uint256 ending) internal {
```

Comment:
Maybe call it `_setProofEnd()`? It could be used to shorten the end time as well, we don't check that the new ending is after the previous ending.